src/communication/default_notifier.rs

```diff
@@ -0,0 +1,338 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+
+use crate::{LocalUriProvider, UListener, UMessageBuilder, UTransport, UUri};
+
+use super::{
+    apply_common_options, build_message, CallOptions, NotificationError, Notifier,
+    RegistrationError, UPayload,
+};
+
+pub struct SimpleNotifier {
+    transport: Arc<dyn UTransport>,
+    uri_provider: Arc<dyn LocalUriProvider>,
+}
+
+impl SimpleNotifier {
+    pub fn new(transport: Arc<dyn UTransport>, uri_provider: Arc<dyn LocalUriProvider>) -> Self {
+        SimpleNotifier {
+            transport,
+            uri_provider,
+        }
+    }
+}
+
+#[async_trait]
+impl Notifier for SimpleNotifier {
```

Comment:
Should we separate `notify` and `start_listening` & `stop_listening` to different traits? If I understand correctly, `notify` is only used by the sender, while the other two are used by the receiver.

Reply:
I have taken inspiration from the [up-java implementation](https://github.com/eclipse-uprotocol/up-java/blob/main/src/main/java/org/eclipse/uprotocol/communication/Notifier.java) where everything is in a single interface. Do you think it has a lot of value to have separate traits here?

Reply:
Hmm... Having different traits is clearer, but I prefer to have a more consistent API within different languages. So, it makes sense to me to have the same way as up-java does.